src/EditorFeatures/Core.Wpf/InlineParameterNameHints/InlineParamNameHintsTagger.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Tagging;
+
+namespace Microsoft.CodeAnalysis.Editor.InlineParameterNameHints
+{
+    /// <summary>
+    /// The purpose of this tagger is to convert the <see cref="InlineParamNameHintDataTag"/> to
+    /// the <see cref="InlineParamNameHintsTag"/>, which actually creates the UIElement. It reacts to
+    /// tags changing and updates the adornments accordingly.
+    /// </summary>
+    internal sealed class InlineParamNameHintsTagger : ITagger<IntraTextAdornmentTag>, IDisposable
+    {
+        private readonly ITagAggregator<InlineParamNameHintDataTag> _tagAggregator;
+        private readonly ITextBuffer _buffer;
+
+        public event EventHandler<SnapshotSpanEventArgs> TagsChanged;
+
+        public InlineParamNameHintsTagger(ITextBuffer buffer, ITagAggregator<InlineParamNameHintDataTag> tagAggregator)
+        {
+            _buffer = buffer;
+            _tagAggregator = tagAggregator;
+            _tagAggregator.TagsChanged += OnTagAggregatorTagsChanged;
+        }
+
+        private void OnTagAggregatorTagsChanged(object sender, TagsChangedEventArgs e)
+        {
+            var spans = e.Span.GetSpans(_buffer);
+            foreach (var span in spans)
+            {
+                TagsChanged?.Invoke(this, new SnapshotSpanEventArgs(span));
+            }
+        }
+
+        public IEnumerable<ITagSpan<IntraTextAdornmentTag>> GetTags(NormalizedSnapshotSpanCollection spans)
+        {
+            var tagsList = new List<ITagSpan<IntraTextAdornmentTag>>();
```

Comment:
Can be done later but we can probably convert this to a iterator with 'yield return'.